waku/v2/protocol/filter/client.go

```diff
@@ -206,6 +236,13 @@ func (wf *WakuFilterLightNode) request(ctx context.Context, params *FilterSubscr
 
 // Subscribe setups a subscription to receive messages that match a specific content filter
 func (wf *WakuFilterLightNode) Subscribe(ctx context.Context, contentFilter ContentFilter, opts ...FilterSubscribeOption) (*SubscriptionDetails, error) {
+	wf.Lock()
```

Comment:
I am wondering having a lock at this high level would slow down the API calls if multiple routines want to invoke them. Because ideally we would want the app to use these API's in multiple routines and not wait for 1 API to finish (which includes sending message out), rather only lock parts of code which we feel would cause concurrency issues.
We should have fine-grained locking as below to address this:
1. To protect only start/stop 
2. Have a separate lock to protect only subscriptionsMap access. Similar to what @harsh-98 has done for peerMap in serviceSlots #684 
I don't think we have any other data that we want to protect other than these 2.

Reply:
Hm, the problem I see is what happens if there is a race condition when someone calls Subscribe/Unsubscribe and Stop at same time, which was the initial issue that originated this PR. 

I'm thinking that perhaps I should change the lock type to an RLock with the idea of allowing concurrent reads, since the lock of `WakuFilterLightNode` is intended to protect only the `started` attribute